app/src/main/java/xyz/malkki/neostumbler/ui/screens/MapScreen.kt

```diff
@@ -284,6 +300,43 @@ fun MapScreen(mapViewModel: MapViewModel = viewModel()) {
     }
 }
 
+private fun addCoverageLayer(style: Style, tileJson: String) {
+    getTileJsonLayerIds(tileJson) { layerIds ->
+        Handler(Looper.getMainLooper()).post {
```

Comment:
This Handler feels a little bit awkward here. I think it would be better to load the layer IDs with coroutines in the view model and then just add them to the style here

Reply:
I done it.